app/services/variants_stock_levels.rb

```diff
@@ -34,12 +34,21 @@ def notify_bugsnag(order, requested_variant_ids, order_variant_ids)
   def variant_stock_levels(line_items)
     Hash[
       line_items.map do |line_item|
-        [line_item.variant.id,
+        variant = scoped_variant(line_item.order.distributor, line_item.variant)
+
+        [variant.id,
          { quantity: line_item.quantity,
            max_quantity: line_item.max_quantity,
-           on_hand: line_item.variant.on_hand,
-           on_demand: line_item.variant.on_demand }]
+           on_hand: variant.on_hand,
+           on_demand: variant.on_demand }]
       end
     ]
   end
+
+  def scoped_variant(distributor, variant)
+    return variant if distributor.blank?
+
+    OpenFoodNetwork::ScopeVariantToHub.new(distributor).scope(variant)
```

Comment:
I think you need to memoize the scoper like this
```ruby
  def scoper(distributor)
    @scoper ||= OpenFoodNetwork::ScopeVariantToHub.new(distributor)
  end
```
this will avoid loading all the hubs overrides several times:
```ruby
  class ScopeVariantToHub
    def initialize(hub, variant_overrides = nil)
      @hub = hub
      @variant_overrides = variant_overrides || VariantOverride.indexed(@hub)
    end
```